packages/aws-cdk-lib/aws-appsync/lib/auth-config.ts

```diff
@@ -0,0 +1,230 @@
+import { Construct } from 'constructs';
+import { CfnApiKey } from './appsync.generated';
+import { IUserPool } from '../../aws-cognito';
+import { IFunction } from '../../aws-lambda';
+import { Duration, Expiration } from '../../core';
+
+/**
+ * enum with all possible values for AppSync authorization type
+ */
+export enum AuthorizationType {
```

Comment:
The authorization config is largely shared between GraphQL and Events, so it has been extracted into a separate file.